test/e2e/test/elasticsearch/steps_mutation.go

```diff
@@ -192,8 +194,16 @@ func (hc *ContinuousHealthCheck) Start() {
 				// recreate the Elasticsearch client at each iteration, since we may have switched protocol from http to https during the mutation
 				client, err := hc.esClientFactory()
 				if err != nil {
-					// treat client creation failure same as unavailable cluster
-					hc.AppendErr(err)
+					// according to https://github.com/kubernetes/client-go/blob/fb61a7c88cb9f599363919a34b7c54a605455ffc/rest/request.go#L959-L960,
+					// client-go requests may return *errors.StatusError or *errors.UnexpectedObjectError, or http client errors.
```

Comment:
Intuitively I think we would like to catch only a small set of known errors _(mostly the one you mentioned in the description: timeout, connection refused and dns related errors)_ but letting all the other ones bubble up. Could you add a comment to explain why we are doing the opposite here ?

Reply:
I think the only reason I did it this way is that those errors [are not trivial to catch](https://stackoverflow.com/questions/22761562/portable-way-to-detect-different-kinds-of-network-error-in-golang), catching expected api errors is easier. 

@barkbay do you think we should change the approach to indeed catch those network errors explicitly, or is adding a comment explaining the reverse approach enough?

Reply:
>  or is adding a comment explaining the reverse approach enough

👍 , I think I'm fine with your proposal, I would just add a comment to understand later why we made this choice.